aiida/cmdline/commands/cmd_import.py

```diff
@@ -25,6 +26,52 @@
 COMMENT_MODE = ['newest', 'overwrite']
 
 
+class ImportPath(click.Path):
+    """AiiDA Verdi implementation of Click's Path-type.
+    Introduce tests to distinguish whether the path is to a local file
+    or a URL string to a file that needs to be downloaded.
+    """
+
+    def __init__(self, **kwargs):
+        super(ImportPath, self).__init__(**kwargs)
+
+    def __call__(self, value, param=None, ctx=None):
+        if value is not None:
+            try:
+                res = super(ImportPath, self).convert(value, param, ctx)
```

Comment:
since you are reusing the `convert` - perhaps it would be cleaner to simply overwrite `convert` directly?

i.e.
```
def convert(..):
   try:
       return super(ImportPath, self).convert(value, param, ctx)
    except click.exceptions.BadParameter:
       new code...
```

the way you do it currently, if somebody were to use your new `convert` method, the result would not be the expected one (since it only handles the URL case)

Reply:
True, This way I don't have to overwrite the `__call__` function as well.

Reply:
Done